### DISCUSSION TITLE
In Next.js 13 app directory, how do I incrementally generate new pages?

### DISCUSSION THREAD

**Discussion by BurgerBurglar** [vercel/next.js#45581] [Help]
I have multiple items from a CMS. `/items/1` all the way to `/items/9999`. The content is immutable, so I don't have to worry about `revalidate`ing them.

However, items do get added to the CMS frequently, maybe multiple times in a day. I want to make a static website. How can I add new static pages incrementally? 

The CMS isn't handled by me, so there's no way I can add a hook.

**Comment by icyJoseph** [comment]
IIRC you have to use the `dynamic params` option: https://beta.nextjs.org/docs/api-reference/segment-config#dynamicparams

> - true (default): Dynamic segments not included in generateStaticParams are generated on demand.
> - false: Dynamic segments not included in generateStaticParams will return a 404.

```ts
export const dynamicParams = true // true | false,
```

If it doesn't work, or it is unstable, you gotta be mindful that appDir is in beta stage, which means some stuff is not quite there yet.

**Comment by minyan-gu** [comment]
For those pages generated on demand (not covered by `generateStaticParams`), seems to me that are not following the `revalidation` config. 

e.g. on this dynamic route, I have set TTL to 300 seconds
```export const revalidate = 300```

However, when I fire subsequent requests to those on demand generated page, the cache status is always marked as `stale` which triggers the background worker to fetch data for every request.

Example response headers recevied
```
age: 7
x-matched-path: /items/[id]
x-powered-by: Next.js
x-vercel-cache: STALE
```

has anyone else encountered this? Feels more like a bug to me, might be the `x-matched-path` is wrong, so it should not have the variable `[id]` within?

Next.js version: `13.3.0`
